controlplane/kubeadm/controllers/kubeadm_control_plane_controller_test.go

```diff
@@ -444,7 +444,10 @@ func TestReconcileClusterNoEndpoints(t *testing.T) {
 		Log:                log.Log,
 		remoteClientGetter: fakeremote.NewClusterClient,
 		recorder:           record.NewFakeRecorder(32),
-		managementCluster:  &internal.ManagementCluster{Client: fakeClient},
+		managementCluster: &fakeManagementCluster{
+			ManagementCluster: &internal.ManagementCluster{Client: fakeClient},
```

Comment:
These look pretty weird. In theory we should have two different fake clients if we're being honest, but I don't think it's a huge deal right now. Alternative strategies may arise as components get pulled apart.